tutorial/10_key_features/003_efficient_optimization_algorithms.py

```diff
@@ -0,0 +1,146 @@
+"""
+.. _pruning:
+
+Efficient Optimization Algorithms
+=================================
+
+Optuna enables efficient hyperparameter optimization by
+adopting state-of-the-art algorithms for sampling hyperparameters and
+pruning efficiently unpromising trials.
```

Comment:
The content seem nice in general but I'm not sure about the balance between the explanations about the samplers vs. the pruners. There's a lot of emphasis on the pruners but the samplers are equally if not more important. Can we make the explanations about these two concepts more equal by e.g. structuring this tutorial as follow?

```
# Sampling and Pruning
# Sampling
## Algorithm
## Select Samplers
# Pruning
## Algorithm
## Activating Pruners
# Which Sampler and Pruner Should be Used?
```

Reply:
In Select Samplers section, do you expect that section to have some executable snippet as well as `activating pruners`?

Reply:
If you meant the following, then yup.
- Select Samplers: Includes executable/example code constructing and specifying sampler 
- Activating Pruners: Includes executable/example code constructing, specifying and using pruner (in the objective)

Reply:
Still not quite sure what you expect but I added the section.